src/software/multithreading/observer_subject_adapter.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <functional>
+
+#include "software/multithreading/observer.h"
+#include "software/multithreading/subject.h"
+
+/**
+ * ObserverSubjectAdapter is a class made to "adapt" a type emitted by a Subject to
```

Comment:
could like this design pattern link: https://refactoring.guru/design-patterns/adapter